Improve error handling in Niko Home Control config flow

VandeurenGlenn opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Adds specific handling for TimeoutError and OSError during controller connection attempts, logging appropriate messages and returning 'cannot_connect'. Updates strings.json to include a 'connection_timeout' error message.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.

  AI tools are welcome, but contributors are responsible for *fully*
  understanding the code before submitting a PR.
-->

- [x] I understand the code I am submitting and can explain how it works.
- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.
- [ ] Any generated code has been carefully reviewed for correctness and compliance with project standards.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

VandeurenGlenn committed:
Improve error handling in Niko Home Control config flow

Adds specific handling for TimeoutError and OSError during controller connection attempts, logging appropriate messages and returning 'cannot_connect'. Updates strings.json to include a 'connection_timeout' error message.

Markus98 commented on homeassistant/components/niko_home_control/config_flow.py:
```diff
@@ -28,8 +28,14 @@ async def test_connection(host: str) -> str | None:
     except Exception:
-        _LOGGER.exception("Unexpected exception")
+        _LOGGER.exception("Unexpected exception during connection")
         return "cannot_connect"
```
`unknown` here?

VandeurenGlenn replied:
Exactly!

Markus98 reviewed (changes requested):
I expect that this PR is a work in progress since it's a draft, but I'm leaving these comments here anyways since they should be easy fixes.

MartinHjelmare reviewed (commented):
Please add tests for the new error cases. The config flow requires 100% test coverage.

Example:

https://github.com/home-assistant/core/blob/76c6b929108d47dcbed13f0029b073740c84cf70/tests/components/niko_home_control/test_config_flow.py#L39-L68

VandeurenGlenn committed:
Refactor error handling in Niko Home Control config flow

Replaces logging-based error handling with explicit error codes for timeout, cannot connect, and unknown errors in the Niko Home Control config flow. Updates string keys and adds/adjusts tests to cover new error cases and ensure correct error reporting.

VandeurenGlenn committed:
Update test docstrings for error cases

Corrected docstrings in test_timeout_connect and test_unknown to accurately describe the errors being tested (timeout and unknown errors, respectively).

VandeurenGlenn committed:
Refactor error and abort keys in strings.json

Reorganized the placement of 'error' and 'abort' keys in the Niko Home Control strings.json to improve structure and avoid duplication.

joostlek committed:
Fix
